core/pkg/fixhandler/fixhandler.go

```diff
@@ -208,6 +208,12 @@ func (h *FixHandler) ApplyChanges(ctx context.Context, resourcesToFix []Resource
 			continue
 		}
 
+		// Handle the case where the resource file starts with ---
+		// causes yaml.Node to misinterpret the resources
+		if fileAsString[:3] == "---" {
+			fileAsString = "# " + fileAsString
+		}
+
```

Comment:
issue: YAML file sanitization is not abstracted away

A function should not deal with implementation details of more than one logical thing at once. This allows it to be testable, maintainable and reduce the amount of things a reader has to know when looking at a function. That is, improve local reasoning.

However, the suggested code tries to do something seemingly small and clever on a low level of abstraction in a function that deals with a high level of abstraction. In other words, before this change, the `ApplyChanges` method knew how to fix resources using a set of lower-level functions that handled the details. It did one thing and, ideally, did it well.

After this change, this method would have to know how to fix resources AND what are the exact steps to sanitize a YAML file so `yaml/v3` will not complain. But there’s no need for this.

We would benefit from having a separate function / method / struct that sanitizes an input file. That way, we can test this separate function and be confident that it fulfills the contract it provides. So `ApplyChanges` can utilize, say, `SanitizeYAML` without knowing how to actually sanitize it.